### PR TITLE
feat: [sc-23880] Sharepoint authentication

### PR DESCRIPTION
**Pull request by jazz-b**
Story details: https://app.shortcut.com/wmfs/story/23880

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -20,71 +19,34 @@ describe('Tests', function () {
...
+      process.env.SHAREPOINT_TENANT_ID &&
+      process.env.SHAREPOINT_DIR_PATH &&
+      process.env.SHAREPOINT_URL
     )) {
       console.log('Missing environment variables, skipping tests.')
       this.skip()
     }
   })
 
-  it('attempt to create a new Sharepoint without passing url', () => {
```
[the remark this answers is not in the document]
Comment (jazz-b) [reply test/test.js:33]: Restored test to check for missing site url.

File: lib/index.js [hunk 2]
```diff
@@ -1,217 +1,354 @@
...
-    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16)
-  })
-}
-
+// see https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins/get-to-know-the-sharepoint-rest-service?tabs=csom
 class Sharepoint {
-  constructor (url) {
-    if (!url) {
-      throw new Error('You must provide a url.')
+  /**
```
Comment (reecebrend) [inline lib/index.js:10]: Can this comment be removed? It's already in the readme and just pollutes the code being in here too

File: test/test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (reecebrend) [reply test/test.js:33]: It'd probably be sensible for us to include checks for the other env vars to ensure the error messages behave as expected

File: lib/index.js (hunk 2 above)
Comment (jazz-b) [reply lib/index.js:10]: Tabled of env vars removed.

**Review by reecebrend — APPROVED** [review]

**Review by meenahoda — APPROVED** [review]

**Pull request merged**